plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -572,17 +655,28 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     private fun startDragging(annotation: T): Boolean {
       if (annotation.isDraggable) {
         dragListeners.forEach { it.onAnnotationDragStarted(annotation) }
-        draggedAnnotation = annotation
+        draggingAnnotation = annotation
+        // Delete the dragging annotation from original source
+        delete(annotation)
+        delegateProvider.getStyle { style ->
+          // Add the dragging annotation to drag layer
+          updateDragSource(style)
+        }
         return true
       }
       return false
     }
 
     private fun stopDragging() {
-      draggedAnnotation?.let { annotation ->
+      draggingAnnotation?.let { annotation ->
         dragListeners.forEach { it.onAnnotationDragFinished(annotation) }
+        draggingAnnotation = null
+        addAnnotation(annotation)
```

Comment:
choose better naming smth like `restoreAnnotationOnDragEnd`

Reply:
This is corresponding to the upper `delete` function. The actual behavior is to add a deleted annotation, might not be constrained to drag. So IMO `addAnnotation` is a better name.